### PR TITLE
EN-12938 Add cookie auth support

### PR DESCRIPTION
**Pull request by msimav**

### CODE REVIEW

File: soda-consumer-scala/src/main/scala/com/socrata/http/impl/EnrichedRequestBuilder.scala [hunk 1]
```diff
@@ -20,6 +19,8 @@ class EnrichedRequestBuilder[T <: RequestBuilderBase[T]](b: T) {
           setUsePreemptiveAuth(true).
           build()
         b.setRealm(realm).addHeader("X-App-Token", appToken)
+      case CookieAuth(cookie) =>
+        b.addCookie(new Cookie(null, "_core_session_id", cookie, null, -1, false))
```
Comment (msimav) [inline soda-consumer-scala/src/main/scala/com/socrata/http/impl/EnrichedRequestBuilder.scala]: I am not sure if passing only `_core_session_id` is enough

**Comment by chrismetcalf** [comment]
@rjmac Can you please take a quick look at this one for me?

File: soda-consumer-scala/src/main/scala/com/socrata/http/impl/EnrichedRequestBuilder.scala (hunk 1 above)
Comment (rjmac) [inline soda-consumer-scala/src/main/scala/com/socrata/http/impl/EnrichedRequestBuilder.scala]: It is

**Comment by rjmac** [comment]
This will work as far as it goes, but it doesn't look like there's a way to get the cookie in the first place?  Also each request you make will send back a new cookie to use with an extended expiration.

**Comment by msimav** [comment]
This is a library and it is usually library users' job to obtain a client cookie. So I don't think we should implement a way to get cookie. We may offer cookiejar support if underlying client let us use one.

**Comment by msimav** [comment]
@rjmac, @chrismetcalf  Can you merge the PR since I don't have write permission to the repo?

**Pull request merged**